tests/unit/test_charm.py

```diff
@@ -20,14 +20,18 @@ def setUp(self):
         self.addCleanup(self.harness.cleanup)
         self.harness.begin()
 
+    @patch("charm.NginxIngressCharm._report_ingress_ips")
     @patch("charm.NginxIngressCharm._report_service_ips")
     @patch("charm.NginxIngressCharm._define_ingress")
     @patch("charm.NginxIngressCharm._define_service")
-    def test_config_changed(self, _define_service, _define_ingress, _report_service_ips):
+    def test_config_changed(
+        self, _define_service, _define_ingress, _report_service_ips, _report_ingress_ips
+    ):
         """Test our config changed handler."""
         # First of all test, with leader set to True.
         self.harness.set_leader(True)
-        _report_service_ips.return_value = ["10.0.1.12"]
+        _report_ingress_ips.return_value = ["10.0.1.12"]
+        _report_service_ips.return_value = ["10.0.1.13"]
```

Comment:
Is this second line needed? It seems that the `return_value` is updated twice in a row without using it in between?

Reply:
Yes, it is needed for the AssertEqual you can find at the end of the test with "confirm the status is as expected" as a comment.

Reply:
Also, the return value is updated for two different things - once for ingress_ips and once for service_ips :)